Nos/Models/PreviewEventRepository.swift

```diff
@@ -0,0 +1,50 @@
+import CoreData
+
+/// Provides a mechanism to store and retrieve an `Event` meant to be used to
+/// preview a note the user is composing.
+protocol PreviewEventRepository {
+    /// Creates a preview event object in the database.
+    /// - Parameters
+    ///   - jsonEvent: JSONEvent object that holds the preview data.
+    ///   - context: Managed object context that will hold the Preview event.
+    func createPreviewEvent(from jsonEvent: JSONEvent, in context: NSManagedObjectContext) throws -> Event?
+
+    /// Deletes the preview event object from the database.
+    /// - Parameters
+    ///   - event: Preview event to delete from the database.
+    ///   - context: Managed object context that holds the Preview event.
+    func deletePreviewEvent(_ event: Event, in context: NSManagedObjectContext) throws
+}
+
+/// Uses `CoreData` to store and retrieve an `Event` meant to be used to
+/// preview a note the user is composing.
+struct DefaultPreviewEventRepository: PreviewEventRepository {
+    /// Creates a preview event object in the database.
+    /// - Parameters
+    ///   - jsonEvent: JSONEvent object that holds the preview data.
+    ///   - context: Managed object context that will hold the Preview event.
```

Comment:
I've generally avoided writing doc comments for `func`s that are documented in the protocol since I think Xcode just shows the protocol docs in the documentation popup. No real issue here other than needing to keep them in sync if anything changes -- just something to keep in mind for the future.